include/okapi/api/units/RQuantityName.hpp

```diff
@@ -0,0 +1,37 @@
+#include "okapi/api/units/QAngle.hpp"
+#include "okapi/api/units/QLength.hpp"
+#include "okapi/api/units/QSpeed.hpp"
+#include <stdexcept>
+#include <typeindex>
+#include <unordered_map>
+
+#pragma once
+
+namespace okapi {
+
+template <class QType> std::string getShortUnitName(QType q) {
+  const std::unordered_map<std::type_index, std::unordered_map<double, const char*>> shortNameMap =
+    {{typeid(meter),
+      {{meter.getValue(), "m"},
+       {decimeter.getValue(), "dm"},
+       {centimeter.getValue(), "cm"},
+       {millimeter.getValue(), "mm"},
+       {kilometer.getValue(), "km"},
+       {inch.getValue(), "in"},
+       {foot.getValue(), "ft"},
+       {yard.getValue(), "yd"},
+       {mile.getValue(), "mi"},
+       {tile.getValue(), "tile"},
+      }},
+     {typeid(degree), {{degree.getValue(), "deg"}, {radian.getValue(), "rad"}}}};
+
+  try {
```

Comment:
Should I make a cpp file for this, or should I leave this in hpp?
If making a cpp, should the map be in cpp or hpp?

Reply:
No, this is fine.